File: Coding Interviews/02 - Basic Knowledge/E366.Fibonacci.py
class Solution:
    def __init__(self):
        self.nums = [0, 1]

    """
    @param n: an integer
    @return: an ineger f(n)
    """
    def fibonacci(self, n):
        # write your code here
        if n < 1:
            return 0

        for i in range(len(self.nums), n):
            self.nums.append(self.nums[i - 2] + self.nums[i - 1])

        return self.nums[n - 1]

File: Coding Interviews/02 - Basic Knowledge/E366.Fibonacci.java
public class Solution {
    /**
     * @param n: an integer
     * @return: an ineger f(n)
     */
    public int fibonacci(int n) {
        // write your code here
        int first = 0, second = 1;
        for (int i = 1; i < n; i++) {
            int third = first + second;
            first = second;
            second = third;
        }

        return first;
    }
}
